### ISSUE TITLE
Validation fails if Dockerfile starts with BOM

### ISSUE THREAD

**Issue by fhemberger** [hadolint/hadolint#335]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Linting should work with Dockerfiles created on Windows, starting with a Byte Order Mark (BOM).

### Actual behavior

```
Dockerfile:1:1 unexpected 'hadolint: <stdout>: commitBuffer: invalid argument (invalid character)
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
1.17.1
```

Using `dos2unix Dockerfile` fixes this issue, but either hadolint should be fault tolerant in this case or produce a better error message.
